woocommerce/woocommerce-android
# Replace deprecated menu usages

Closes: #7866
Closes: #7207


### Description

The PR removes left-behind `super.menu` calls as well as menu-related unused code in `BaseAddressEditingFragment`.
Suppresses/fixes warnings in the tests

Also, it makes warnings as errors for the compiler

### Testing instructions

* Open order details
* Modify shipping and billing address
* Notice that the "done" menu button works the same as before

### Images/gif


https://user-images.githubusercontent.com/4923871/202118683-8821ddb1-01df-4d4b-8174-14a66924cdcd.mp4
